java/src/main/java/ai/rapids/cudf/CuFile.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.rapids.cudf;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+
+/**
+ * JNI wrapper for accessing the cuFile API.
+ */
+public class CuFile {
+  private static final Logger log = LoggerFactory.getLogger(CuFile.class);
+  private static boolean initialized = false;
+  private static long driverPointer = 0;
+
+  static {
+    initialize();
+  }
+
+  /**
+   * Load the native libraries needed for libcufilejni, if not loaded already; open the cuFile
+   * driver, and add a shutdown hook to close it.
+   */
+  private static synchronized void initialize() {
+    if (!initialized) {
+      try {
+        NativeDepsLoader.loadNativeDeps(new String[]{"cufilejni"});
+        driverPointer = createDriver();
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+          destroyDriver(driverPointer);
+        }));
+        initialized = true;
+      } catch (Throwable t) {
+        log.error("Could not load cuFile jni library...", t);
+      }
+    }
+  }
+
+  private static native long createDriver();
+
+  private static native void destroyDriver(long pointer);
+
+  /**
+   * Copy a device buffer to a given file path.
+   *
+   * @param path The file path to copy to.
+   * @param buffer The device buffer to copy from.
+   * @param append Whether to append to the file.
+   * @return The file offset from which the buffer was appended.
+   */
+  public static long copyDeviceBufferToFile(File path, DeviceMemoryBuffer buffer, boolean append) {
+    return copyToFile(path.getAbsolutePath(), buffer.getAddress(), buffer.getLength(), append);
```

Comment:
nit: From reading through the native implementations and the cuFile APIs it does not appear that these APIs are strictly thread safe.  It might be good to document that you might get errors if trying to access the same file from multiple different threads. I was a little concerned that append might result in data corruption, but it should not if `cuFileHandleRegister` prevents multiple handles to the same file from happening.

Reply:
Right, these methods are not thread safe if the file path points to the same file on disk. Added to the javadoc.